### DISCUSSION TITLE
How to build and use RAUC 1.7+ on Debian bullseye

### DISCUSSION THREAD

**Discussion by martin-petzold** [rauc/rauc#1130] [Q&A] [answered]
I would like to use a more recent RAUC on my Debian bullseye platform. We would like to make use of streaming installation. However, in the Debian bullseye repository I only find version 1.5.1 [1]. Now I could try to compile RAUC from source, but it seems to require a newer libc6 and libglib2.0 version as well as libssl3, which are not available in Debian bullseye repository. I compare here with 1.8 available in Debian bookwork repository [2].

How can I build and use RAUC 1.7+ on Debian bullseye platform?

[1] https://packages.debian.org/bullseye/rauc
[2] https://packages.debian.org/bookworm/rauc

**Comment by ejoerns** [comment]
These are just the library versions that are available in *bookworm*. I don't see anything that should prevent building RAUC under *bullseye*. According to my test, it even builds on *buster*. The only limitation there is that meson builds do not work because of a too-old meson version.

https://github.com/ejoerns/rauc/actions/runs/4983297591/jobs/8920132932

**Reply by martin-petzold** [reply]
And what about the libssl3 dependency? This library is not available for Debian bullseye...

**Reply by ejoerns** [reply, marked as answer]
Where do you see such a dependency? bookworm comes with openssl 3.0 und thus it builds against this. On bullseye we have openssl 1.1.1 and thus it builds against this instead. Both are more or less API compatible and RAUC supports both.

The actual dependency is defined here:

https://github.com/rauc/rauc/blob/master/meson.build#L40

**Reply by martin-petzold** [reply]
Okay thanks for the link to the build requirements.
